aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -0,0 +1,110 @@
+  return at::ceil_out(result, self);
+}
+Tensor floor(const Tensor &self) {
+  Tensor result = self.type().tensor();
+  return at::floor_out(result, self);
+}
+Tensor round(const Tensor &self) {
+  Tensor result = self.type().tensor();
+  return at::round_out(result, self);
+}
+Tensor trunc(const Tensor &self) {
+  Tensor result = self.type().tensor();
+  return at::trunc_out(result, self);
+}
+Tensor sqrt(const Tensor &self) {
+  Tensor result = self.type().tensor();
+  return at::sqrt_out(result, self);
+}
+
+Tensor &ceil_(Tensor &self) { return at::ceil_out(self, self); }
+Tensor &floor_(Tensor &self) { return at::floor_out(self, self); }
+Tensor &round_(Tensor &self) { return at::round_out(self, self); }
+Tensor &trunc_(Tensor &self) { return at::trunc_out(self, self); }
+Tensor &sqrt_(Tensor &self) { return at::sqrt_out(self, self); }
+
+// \WRAP OPS #################################################################
+
+bool _unops_out_cpu(unary_type *f, Tensor &result, const Tensor &self) {
+  if (result.is_contiguous() && self.is_contiguous()) {
+    result.resize_(self.sizes());
+    f(result, self);
+    return true;
+  }
+  return false;
+}
+
+// CPU OPS ###################################################################
+
+Tensor &_ceil_out_cpu(Tensor &result, const Tensor &self) {
+  return _unops_out_cpu(ceilImpl, result, self) ? result
+                                                : at::_ceil_out(result, self);
+}
+Tensor &_floor_out_cpu(Tensor &result, const Tensor &self) {
+  return _unops_out_cpu(floorImpl, result, self) ? result
+                                                 : at::_floor_out(result, self);
+}
+Tensor &_round_out_cpu(Tensor &result, const Tensor &self) {
+  return _unops_out_cpu(roundImpl, result, self) ? result
+                                                 : at::_round_out(result, self);
+}
+Tensor &_trunc_out_cpu(Tensor &result, const Tensor &self) {
+  return _unops_out_cpu(truncImpl, result, self) ? result
+                                                 : at::_trunc_out(result, self);
+}
+Tensor &_sqrt_out_cpu(Tensor &result, const Tensor &self) {
+  return _unops_out_cpu(sqrtImpl, result, self) ? result
+                                                : at::_sqrt_out(result, self);
+}
+
+// \CPU OPS #################################################################
+
+// CUDA OPS #################################################################
+
+Tensor &_ceil_out_cuda(Tensor &result, const Tensor &self) {
+  return at::_ceil_out(result, self);
+}
+Tensor &_floor_out_cuda(Tensor &result, const Tensor &self) {
+  return at::_floor_out(result, self);
+}
+Tensor &_round_out_cuda(Tensor &result, const Tensor &self) {
+  return at::_round_out(result, self);
+}
+Tensor &_trunc_out_cuda(Tensor &result, const Tensor &self) {
+  return at::_trunc_out(result, self);
+}
+Tensor &_sqrt_out_cuda(Tensor &result, const Tensor &self) {
+  return at::_sqrt_out(result, self);
+}
```

Comment:
This looks like a lot of boilerplate per op. Can't we use a macro to generate those 3 functions?

Reply:
Working on this. I'm trying to avoid macros and use templates instead, but maybe I should just give up on that.

Reply:
I think you're right about the boilerplate, but let's not worry about putting it in a macro yet. I think the code pattern will likely change very soon and it's often harder to change these things inside a macro than in the explicit definitions.

Reply:
The fact that it's going to change soon is only another argument for using the macro! There's no point in doing a ton of manual changes when you can just modify a single macro